Loosen restriction on optional arguments in CCPP Framework

dustinswales opened the pull request:
<!-- INSTRUCTIONS: 
- PLEASE READ/FOLLOW THE DIRECTIONS IN EACH SECTION
- Complete the 'Commit Queue Requirements' below
- Please use github markup as much as possible (https://docs.github.com/en/get-started/writing-on-github)
- Please leave your PR in a draft state until all underlying work is completed.
-->
## Commit Queue Requirements:
<!--
- Please complete the items that follow this.
- Please "check off" completed items. Use [X] for a filled in checkbox or leave it [ ] for an empty checkbox
- Your PR will not be considered until all requirements are met.
- THIS IS YOUR RESPONSIBILITY
 -->
- [ ] Fill out all sections of this template.
- [ ] All sub component pull requests have been reviewed by their code managers.
- [ ] Run the full Intel+GNU RT suite (compared to current baselines) on either Hera/Derecho/Hercules
- [ ] Commit 'test_changes.list' from previous step
---
## Description:
<!--
Loosen restriction on optional arguments in CCPP.
This PR contains changes to the CCPP Framework ([PR627](https://github.com/NCAR/ccpp-framework/pull/627)) and CCPP Physics ([PR248](https://github.com/ufs-community/ccpp-physics/pull/248)), reducing the restrictions on optional arguments and corresponding schemes changes, respectively.
-->


### Commit Message:
```
* UFSWM - Loosen restriction on optional arguments in CCPP.
  * FV3 - Loosen restriction on optional arguments in CCPP.
    * ccpp-physics - Loosen restriction on optional arguments in CCPP.
    * ccpp-framework - Loosen restriction on optional arguments in CCPP.
```

### Priority:
* Normal.

## Git Tracking
### UFSWM:
* None

### Sub component Pull Requests:
* FV3: [NOAA-EMC/fv3atm#917](https://github.com/NOAA-EMC/fv3atm/pull/917)
  * ccpp-physics: [ufs-community/ccpp-physics#248](https://github.com/ufs-community/ccpp-physics/pull/248)
  * ccpp-frameowrk:[ NCAR:ccpp-framework#627](https://github.com/NCAR/ccpp-framework/pull/627)

### UFSWM Blocking Dependencies:
* None

## Changes
### Regression Test Changes (Please commit test_changes.list):
* No Baseline Changes.
(Waiting on full set of RTs on Hera, Will update this PR when tests finished)

### Input data Changes:
* None.

### Library Changes/Upgrades:
* No Updates
  
---
<!-- STOP!!! THE FOLLOWING IS FOR CODE MANAGERS ONLY. PLEASE DO NOT FILL OUT -->
## Testing Log:
- RDHPCS
  - [ ] Hera
  - [ ] Orion
  - [ ] Hercules
  - [ ] Jet
  - [ ] GaeaC5
  - [ ] GaeaC6
  - [ ] Derecho
- WCOSS2
  - [ ] Dogwood/Cactus
  - [ ] Acorn
- [ ] CI
- [ ] opnReqTest (complete task if unnecessary)

grantfirl commented:
@dustinswales Do you have any RT logs? I'm guessing this isn't supposed to change baselines? Let me know if you'd like me to run RTs on this.

dustinswales commented:
@grantfirl I'm addressing your issues in the physics and will rerun the RTs.
Thanks for syncing my branches!

grantfirl commented:
> @grantfirl I'm addressing your issues in the physics and will rerun the RTs. Thanks for syncing my branches!

I think that I fixed the inconsistency that I pointed out. I don't think anything else is required on the physics side. Just RTs at this point. You may want to double-check.

grantfirl commented:
Note that the ccpp-framework PR changed to https://github.com/NCAR/ccpp-framework/pull/638 and is already merged.

dustinswales commented:
> > @grantfirl I'm addressing your issues in the physics and will rerun the RTs. Thanks for syncing my branches!
> 
> I think that I fixed the inconsistency that I pointed out. I don't think anything else is required on the physics side. Just RTs at this point. You may want to double-check.

@grantfirl I just saw that. Thanks!
I'll rerun the RTs now and update the logs.

grantfirl commented:
> > > @grantfirl I'm addressing your issues in the physics and will rerun the RTs. Thanks for syncing my branches!
> > 
> > 
> > I think that I fixed the inconsistency that I pointed out. I don't think anything else is required on the physics side. Just RTs at this point. You may want to double-check.
> 
> @grantfirl I just saw that. Thanks! I'll rerun the RTs now and update the logs.

@dustinswales Any update on these RTs?

dustinswales commented:
> > > > @grantfirl I'm addressing your issues in the physics and will rerun the RTs. Thanks for syncing my branches!
> > > 
> > > 
> > > I think that I fixed the inconsistency that I pointed out. I don't think anything else is required on the physics side. Just RTs at this point. You may want to double-check.
> > 
> > 
> > @grantfirl I just saw that. Thanks! I'll rerun the RTs now and update the logs.
> 
> @dustinswales Any update on these RTs?

@grantfirl Some tests are failing and I need to look into this. Sigh.

grantfirl commented:
@dustinswales This looks like it is coming up in the queue. Were you able to figure out the failing tests?

grantfirl commented:
> @dustinswales This looks like it is coming up in the queue. Were you able to figure out the failing tests?

@dustinswales Could you push the RT logs for this in prep for the UFS merge queue?

dustinswales commented:
@grantfirl I ran the tests the other day and they all passed, but the ecflow job didn't finish and create the new RT logs.
You can see here on Hera (grep FAIL
/scratch1/BMC/gmtb/Dustin.Swales/UFS/framework/optional_args/ufs-weather-model-2/tests/logs/log_hera/*log)

I don't know much about ecflow, but is there a way to restart this 99.99999999% completed job to create the logs?
It would be a waste to run the tests again.
@jkbk2004 @BrianCurtis-NOAA Is there a way to finish a stalled ecflow job? Or do you know who I could ask?

BrianCurtis-NOAA commented:
> @grantfirl I ran the tests the other day and they all passed, but the ecflow job didn't finish and create the new RT logs. You can see here on Hera (grep FAIL /scratch1/BMC/gmtb/Dustin.Swales/UFS/framework/optional_args/ufs-weather-model-2/tests/logs/log_hera/*log)
> 
> I don't know much about ecflow, but is there a way to restart this 99.99999999% completed job to create the logs? It would be a waste to run the tests again. @jkbk2004 @BrianCurtis-NOAA Is there a way to finish a stalled ecflow job? Or do you know who I could ask?

It's not lost on me the need for the ability to start the testing process at any point (even at 99.99999999% completed to just regerenate the logs). Unfortunately, that capability does not exist because the env vars that the log generation function needs to complete are lost when the job fails/ends.

Since I've heard this on numerous occasions, i'll see what can be done.

jkbk2004 commented:
@dustinswales can you sync up branch?

grantfirl commented:
> @dustinswales can you sync up branch?

@jkbk2004 @dustinswales Done

dustinswales commented:
@BrianCurtis-NOAA Thanks for the explanation. For me this happens when my session times out during an ecflow job.

@grantfirl Thanks for syncing the branches!

jkbk2004 commented:
@dustinswales can you sync up branch? we can work on this pr.

jkbk2004 commented:
@BrianCurtis-NOAA @FernandoAndrade-NOAA this pr is ready.

jkbk2004 commented:
we can start merging process.

jkbk2004 commented:
@dustinswales fv3 new hash is https://github.com/NOAA-EMC/fv3atm/commit/de5d1c2b2a5039385251de2fd7baf1f6552e8dae

jkbk2004 commented:
Created an MOM6 code management issue: https://github.com/ufs-community/ufs-weather-model/issues/2717
